gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -774,7 +799,12 @@ private SeekableByteChannel open(
     }
 
     return new GoogleCloudStorageReadChannel(
-        storage, resourceId, errorExtractor, clientRequestHelper, readOptions) {
+        storage,
+        storageRequestFactory,
```

Comment:
If both are not necessary, then let's leave just `storageRequestFactory` and even if both are necessary, then better to have a getter for `storage` in the factory instead of passing both, I think.

Reply:
+1, will make the change.